core/src/main/java/com/linecorp/armeria/client/ConnectionPoolMetrics.java

```diff
@@ -32,7 +43,13 @@
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Tag;
 
-final class ConnectionPoolMetrics {
+final class ConnectionPoolMetrics implements SafeCloseable {
+
+    private static final ScheduledExecutorService CLEANUP_EXECUTOR =
```

Comment:
Can we use the blocking task executor similar you did in `k8sEndpointGroup`?

Reply:
I prefer a dedicated thread. We don't know how long the remove() operations takes and as the number of `ConnectionPoolMetrics` increases, the number of jobs will also increases. An isolated environment may be better than a blocking task executor used for handling requests.

Reply:
That is true. But I also don't want to waste resources creating a thread that is used once an hour. 🤔

Reply:
It makes sense. I am convinced.

Reply:
> We don't know how long the remove() operations takes and as the number of ConnectionPoolMetrics increases,

Yeah, but I believe it's really a problem when multiple threads trying to remove each meter from the meter registry.
Now it's done by just one thread, so I think there are not so many contentions.

Reply:
> It makes sense. I am convinced.

Thank you for your understanding. 🙏 
If you want, I can push a commit for you. Or you can do it by yourself after you get back from the day-off. 
I think this is not urgent, so we don't have to hurry. Please get some rest when you day-off. 😆

Reply:
if we use a blocking task executor, multiple tasks may be scheduled depending on the number of ConnectionPoolMetrics. Although the chance of contention would be extremely low.

Reply:
> If you want, I can push a commit for you.

Please push the commit. I think that it would be a trivial change.

Reply:
I don’t want to add additional breaking changes in this PR as we are going to release a patch version for the PR.